internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -367,17 +413,48 @@ func resourceMonitorDiagnosticSettingUpdate(d *pluginsdk.ResourceData, meta inte
 		}
 	}
 
-	// if no logs/metrics are enabled the API "creates" but 404's on Read
+	var metrics []diagnosticsettings.MetricSettings
 	hasEnabledMetrics := false
-	if !hasEnabledLogs {
+
+	if d.HasChange("enabled_metric") {
+		enabledMetrics := d.Get("enabled_metric").(*pluginsdk.Set).List()
+		if len(enabledMetrics) > 0 {
+			expandEnabledMetrics := expandMonitorDiagnosticsSettingsEnabledMetrics(enabledMetrics)
+			if err != nil {
+				return fmt.Errorf("expanding enabled_metric: %+v", err)
+			}
+			metrics = *expandEnabledMetrics
+			hasEnabledMetrics = true
+		} else if existing.Model != nil && existing.Model.Properties != nil && existing.Model.Properties.Metrics != nil {
+			// if the enabled_metric is updated to empty, we disable the metric explicitly
```

Comment:
Just to confirm, is disabling them going to cause any issue? Does removing them work?

Reply:
Disabling didn't cause any problems when I tested the API. There's no way to remove them, Azure always makes them and then toggles `enabled` in lieu of deleting.